sql/catalog.go

```diff
@@ -226,62 +237,3 @@ func suggestSimilarTablesAsOf(db VersionedDatabase, ctx *Context, tableName stri
 	similar := similartext.Find(tableNames, tableName)
 	return ErrTableNotFound.New(tableName + similar)
 }
-
-// sliceDBProvider is a collection of Database.
-type sliceDBProvider []Database
```

Comment:
You should preserve and export this in the sqle package, so that people who only have a DB implementation have an easy path to use the engine